### DISCUSSION TITLE
New cy.session() feature problem in e2e testing

### DISCUSSION THREAD

**Discussion by kpioro** [cypress-io/cypress#21625] [Questions and Help]
Hey, 
Since you are introducing a new cy.session() approach, I'm worried about handling e2e tests. My e2e test scenario looks like this
-  **before()** - visit sign up page
- **it()** - provide customer fname/lname
- **it()** - provide customer address
- etc.

Now with cy.session() introdced, I have to place cy.visit() in each **it()** before firing any custom command to avoid landing in blank:page.
Is there any other way to handle this kind of issue?
In previous version I used only Cypress.Cookies.preserveOnce('_contador_session'); to store cookie while going through application, and it worked just fine.

**Comment by mamtamanu** [comment]
I have the same issue too
I am using "cypress": "^11.0.0"
In my application, for each it() block, I have to place cy.visit() with url+multiple query parameters (finding difficult to provide all the query parameters as endpoints)
while landing each page, shows default blank page for a sec
is there any solution for it?

**Comment by emilyrohrbough** [comment]
@kpioro / @mamtamanu 

We have taken this feedback and [feedback from the Experimental Discussion](https://github.com/cypress-io/cypress/discussions/17887#discussioncomment-2835987) and as at team we evaluated the test isolation behavior and found it wasn't behaving how users expected and have made adjustments.

A `testIsolation` configuration was introduced in 10.5.0 to allow better control of this concern. In [Cypress 10.11.0](https://docs.cypress.io/guides/references/changelog#10-11-0), the way that Cypress handles [test isolation](https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests#Test-Isolation) changed. The previous modes of `legacy` and `strict` have been replaced with `on` and `off`. (). 

- The default mode is `on` when [experimentalSessionAndOrigin](https://docs.cypress.io/guides/references/experiments) is enabled.
- The [`cy.session()`](https://docs.cypress.io/api/commands/session) command now inherits the test isolation behavior for the suite it runs in.

To accomplished the example provided above, you will need to set `testIsolation="off"` in the suite or in the root-level configuration. Be sure to understand there are risks with turning `testIsolation="off"` and it will be on you to ensure your browser context maintains a clean slate for independent tests.
